From 2208abc9487bfab9e1db16c3e98e022e213c63a3 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Wed, 10 Dec 2025 17:12:38 +0800
Subject: [PATCH 62/63] loonggpu: remove gsgpu_driver_lastclose_kms on Linux >=
 6.12

This is not used on Linux >= 6.12, and it fails to build on Linux 6.19.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_kms.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/loonggpu/loonggpu_kms.c b/loonggpu/loonggpu_kms.c
index 42371eb..d34c0a6 100644
--- a/loonggpu/loonggpu_kms.c
+++ b/loonggpu/loonggpu_kms.c
@@ -6,6 +6,7 @@
 #include <linux/slab.h>
 #include <linux/pm_runtime.h>
 #include <linux/pci.h>
+#include <linux/version.h>
 #include <drm/drm_debugfs.h>
 #include "loonggpu_gtt_mgr_helper.h"
 
@@ -536,6 +537,7 @@ static int loonggpu_info_ioctl(struct drm_device *dev, void *data, struct drm_fi
 }
 
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0)
 /*
  * Outdated mess for old drm with Xorg being in charge (void function now).
  */
@@ -551,6 +553,7 @@ void loonggpu_driver_lastclose_kms(struct drm_device *dev)
 	drm_fb_helper_lastclose(dev);
 	vga_switcheroo_process_delayed_switch();
 }
+#endif
 
 /**
  * loonggpu_driver_open_kms - drm callback for open
-- 
2.52.0

